core/src/main/kotlin/renderers/FileWriter.kt

```diff
@@ -15,19 +16,46 @@ class FileWriter(val root: String, override val extension: String): OutputWriter
         createdFiles.add(path)
 
         try {
-//            println("Writing $root/$path$ext")
             val dir = Paths.get(root, path.dropLastWhile { it != '/' }).toFile()
             dir.mkdirsOrFail()
-            Paths.get(root, "$path$ext").toFile().writeText(text)
-        } catch (e : Throwable) {
+            Files.write(Paths.get(root, "$path$ext"), text.lines())
+        } catch (e: Throwable) {
             println("Failed to write $this. ${e.message}")
             e.printStackTrace()
         }
     }
 
+    override fun writeResources(pathFrom: String, pathTo: String) {
+        val rebase = fun(path: String) =
+            "$pathTo/${path.removePrefix(pathFrom)}"
+        val dest = Paths.get(root, pathTo).toFile()
+        dest.mkdirsOrFail()
+        val uri = javaClass.getResource(pathFrom).toURI()
+        val fs = getFileSystemForURI(uri)
+        val path = fs.getPath(pathFrom)
+        for (file in Files.walk(path).iterator()) {
+            if (Files.isDirectory(file)) {
+                val dirPath = file.toAbsolutePath().toString()
+                Paths.get(root, rebase(dirPath)).toFile().mkdirsOrFail()
+            } else {
+                val filePath = file.toAbsolutePath().toString()
+                Paths.get(root, rebase(filePath)).toFile().writeBytes(
+                    javaClass.getResourceAsStream(filePath).readBytes()
+                )
+            }
+        }
+    }
+
     private fun File.mkdirsOrFail() {
         if (!mkdirs() && !exists()) {
             throw IOException("Failed to create directory $this")
         }
     }
+
+    private fun getFileSystemForURI(uri: URI): FileSystem =
+        try {
+            FileSystems.newFileSystem(uri, emptyMap<String, Any>())
```

Comment:
This will fail for regular directories when running tests, see 280859d8c9bdd6d5b5417a97edf1b1378bd9a9c7

Reply:
I've forgotten about this problem. Now the fix is also on this branch.